From 500e186a936a7c8edb1248dc2def6896130b4596 Mon Sep 17 00:00:00 2001
From: Omar Ramirez Luna <omar.r.ramirez.luna@intel.com>
Date: Wed, 12 Oct 2016 14:17:22 -0700
Subject: [PATCH 25/27] intel-mid: split keyboard gpio SFI implementation from
 the driver

soc_button_array driver is unusable because of dependencies on
the platform SFI implementation for intel-mid platforms. These
dependencies on SFI make compilation break for ACPI based
platforms.

Signed-off-by: Omar Ramirez Luna <omar.r.ramirez.luna@intel.com>
---
 arch/x86/platform/intel-mid/device_libs/Makefile | 2 +-
 drivers/platform/x86/Kconfig                     | 9 +++++++++
 2 files changed, 10 insertions(+), 1 deletion(-)

diff --git a/arch/x86/platform/intel-mid/device_libs/Makefile b/arch/x86/platform/intel-mid/device_libs/Makefile
index 2aec947..88b097f 100644
--- a/arch/x86/platform/intel-mid/device_libs/Makefile
+++ b/arch/x86/platform/intel-mid/device_libs/Makefile
@@ -18,5 +18,5 @@ obj-$(subst m,y,$(CONFIG_INPUT_BMA150)) += platform_bma023.o
 obj-$(subst m,y,$(CONFIG_GPIO_PCA953X)) += platform_tca6416.o
 obj-$(subst m,y,$(CONFIG_DRM_MEDFIELD)) += platform_tc35876x.o
 # MISC Devices
-obj-$(subst m,y,$(CONFIG_KEYBOARD_GPIO)) += platform_gpio_keys.o
+obj-$(subst m,y,$(CONFIG_INTEL_MID_KEYBOARD_GPIO)) += platform_gpio_keys.o
 obj-$(subst m,y,$(CONFIG_INTEL_MID_WATCHDOG)) += platform_wdt.o
diff --git a/drivers/platform/x86/Kconfig b/drivers/platform/x86/Kconfig
index 4c1d55d..5a92dfe 100644
--- a/drivers/platform/x86/Kconfig
+++ b/drivers/platform/x86/Kconfig
@@ -764,6 +764,15 @@ config INTEL_MID_POWER_BUTTON
 
 	  If unsure, say N.
 
+config INTEL_MID_KEYBOARD_GPIO
+	tristate "keyboard button driver for Intel MID platforms"
+	depends on INTEL_SCU_IPC && INPUT && KEYBOARD_GPIO
+	help
+	  This driver handles the SOC Array buttons on the Intel MID platforms
+	  using SFI.
+
+	  If unsure, say N.
+
 config INTEL_MFLD_THERMAL
        tristate "Thermal driver for Intel Medfield platform"
        depends on MFD_INTEL_MSIC && THERMAL
-- 
2.9.3

